src/LinearOp/IProjMatrix.cpp

```diff
@@ -11,73 +11,48 @@
 
 #include "LinearOp/IProjMatrix.hpp"
 
-#include "Matrix/VectorEigen.hpp"
-
-#include <Eigen/Core>
-#include <Eigen/Dense>
+#include "geoslib_define.h"
 
 int IProjMatrix::mesh2point(const VectorDouble& inv,
-                           VectorDouble& outv) const
+                                  VectorDouble& outv) const
 {
-    Eigen::Map<const Eigen::VectorXd> myInv(inv.data(), inv.size());
-    outv.resize(getPointNumber());
-    Eigen::VectorXd myOut(outv.size());
-    
-    // Assume outv has the good size
-    int error = mesh2point(myInv, myOut);
-    
-    Eigen::Map<Eigen::VectorXd>(outv.data(), outv.size()) = myOut;
-    return error;
+  outv.resize(getPointNumber());
+  constvect myInv(inv.data(), inv.size());
+  vect myOut(outv);
+  return mesh2point(myInv, myOut);  
 }
 
 int IProjMatrix::point2mesh(const VectorDouble& inv,
                            VectorDouble& outv) const
 {
-    outv.resize(getApexNumber());
-    Eigen::Map<const Eigen::VectorXd> myInv(inv.data(), inv.size());
-    Eigen::VectorXd myOut(outv.size());
-    
-    // Assume outv has the good size
-    int error= point2mesh(myInv, myOut);
-    
-    Eigen::Map<Eigen::VectorXd>(outv.data(), outv.size()) = myOut;
-    return error;
+  outv.resize(getApexNumber());
+  constvect myInv(inv.data(), inv.size());
+  vect myOut(outv);
```

Comment:
Here, you are forced to define local variables of type `std::span` to pass them by reference to `point2mesh`. You can avoid that by passing `std::span` by copy and calling `point2mesh(inv.getVector(), outv.getVector()); `, using the implicit conversion from `std::vector` to `std::span` or using temporaries: `point2mesh(constvect{inv}, vect{outv})`

Reply:
Ok to supress references. I remarked that an automatic conversion was done but are we sure that there is no copy (remember the case of Eigen::Map<Eigen::VectorXd> given in place of Eigen::VectorXd, it worked but you told me that a copy was made).